.github/workflows/project_setup_e2e.yml

```
@@ -0,0 +1,52 @@
+name: Project setup
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18.x
+
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.3.0'
+          bundler-cache: true
+
+      - name: Install Ruby dependencies
+        run: |
+          gem install bundler
+          bundle install
+          
+      - name: Setup application
+        run: | 
+          sudo apt-get install -y expect
+          ./ci-setup.sh
+
+      - name: Start server
+        run: |
+          bin/rails s &
```

You don't need pipe when you are running one command only.
```suggestion
        run: bin/rails s &
```